services/src/main/java/org/keycloak/authorization/authorization/AuthorizationTokenService.java

```diff
@@ -402,19 +426,25 @@ private ResourceServer getResourceServer(PermissionTicketToken ticket, KeycloakA
         String issuedFor = ticket.getIssuedFor();
 
         if (issuedFor == null) {
-            throw new CorsErrorResponseException(request.getCors(), OAuthErrorException.INVALID_REQUEST, "You must provide the issuedFor", Status.BAD_REQUEST);
+            CorsErrorResponseException missingIssuedForException = new CorsErrorResponseException(request.getCors(), OAuthErrorException.INVALID_REQUEST, "You must provide the issuedFor", Status.BAD_REQUEST);
+            fireErrorEvent(request.getEvent(), Errors.MISSING_ISSUED_FOR, missingIssuedForException);
```

Comment:
I would just use Errors.INVALID_REQUEST. The details should be enough to understand what happened.

Reply:
agreed